test/integration/auth/podsecurity_test.go

```diff
@@ -71,6 +87,42 @@ func TestPodSecurityGAOnly(t *testing.T) {
 	podsecuritytest.Run(t, opts)
 }
 
+func TestPodSecurityWebhook(t *testing.T) {
+	// Enable all feature gates needed to allow all fields to be exercised
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ProcMountType, true)()
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.WindowsHostProcessContainers, true)()
+	// The webhook should pass tests even when PodSecurity is disabled.
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.PodSecurity, false)()
+
+	// Start test API server.
+	capabilities.SetForTests(capabilities.Capabilities{AllowPrivileged: true})
+	testServer := kubeapiservertesting.StartTestServerOrDie(t, kubeapiservertesting.NewDefaultTestServerOptions(), []string{
+		"--anonymous-auth=false",
+		"--allow-privileged=true",
+	}, framework.SharedEtcd())
+	t.Cleanup(testServer.TearDownFn)
+
+	webhookAddr, err := startPodSecurityWebhook(t, testServer)
+	if err != nil {
+		t.Fatalf("Failed to start webhook server: %v", err)
+	}
+	if err := installWebhook(testServer.ClientConfig, webhookAddr); err != nil {
+		t.Fatalf("Failed to install webhook configuration: %v", err)
+	}
+
```

Comment:
need to wait for the webhook to become effective before kicking off the test or you'll get flakes (can poll dry-run creating a namespace with malformed labels to wait for the invalid error)

Reply:
The start method waits for readyz, you don't think that's sufficient?

Reply:
no, readyz makes sure the webhook endpoint is ready, this needs to wait for the kube-apiserver to observe the AdmissionConfiguration object and for it to become effective, otherwise the first pod creation races the AdmissionConfiguration observation inside kube-apiserver